src/lib/device/virtual_dev.cpp

```diff
@@ -217,41 +211,66 @@ BlkAllocStatus VirtualDev::do_alloc_blk(blk_count_t nblks, const blk_alloc_hints
     }
 }
 
-BlkAllocStatus VirtualDev::alloc_blk_from_chunk(blk_count_t nblks, const blk_alloc_hints& hints,
-                                                std::vector< BlkId >& out_blkid, Chunk* chunk) {
+BlkAllocStatus VirtualDev::alloc_blks(blk_count_t nblks, blk_alloc_hints const& hints,
```

Comment:
It also needs to be exposed to data service layer for caller to switch the API if alloc_blks with MultiBlkId fails, right?